pkg/api/pin.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2021 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package api
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/ethersphere/bee/pkg/jsonhttp"
+	"github.com/ethersphere/bee/pkg/storage"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/gorilla/mux"
+)
+
+// pinChunk pin's the already created chunk given its address.
+// It fails if the chunk is not present in the local store.
+// It also increments a pin counter to keep track of how many
+// pin requests are originating for this chunk.
+func (s *server) pinChunk(w http.ResponseWriter, r *http.Request) {
```

Comment:
this is no longer needed, since we don't provide explicit chunk level pinning (only if this is done through the chunk upload service or through pinning unpinned roots which we can also use to potentially unpin a single chunk)